src/openrct2/ride/Vehicle.cpp

```diff
@@ -1522,16 +1522,16 @@ bool Vehicle::OpenRestraints()
         }
         if (vehicleEntry->animation == VEHICLE_ENTRY_ANIMATION_OBSERVATION_TOWER && vehicle->animation_frame != 0)
         {
-            if (vehicle->animationState + 0x3333 < 0xFFFF)
+            if (vehicle->animationState <= 0xCCCC)
```

Comment:
To me, it looks like the old code checked for `< 0xCCCC`, not `<=`.

Reply:
The previous code relied on overflow to reset the count. This now does not. During the conversion the two branches of the if were flipped. If you look at the other animation modes in this function you'll see how they now operate in the same way. Basically on the frame where it resets to zero we want the animation to do something (change frame) all other times just count up.

Reply:
The change from `< 0xCCCC` to `<= 0xCCCC` goes hand-in-hand with resetting the timer to 0. Since `0x1000 % 0x3333 != 0`, there state is constantly 'slipping' backwards. This normally isn't a problem with the highlighted code unless the number of frames the animation has cycled through is a multiple of 0x3333, in which case the above code will update the car after 5 frames of waiting instead of 4. Since the new behavior of resetting animationState to 0 will always trigger that behavior, changing it to include 0xCCCC is necessary.